public/app/core/components/ForgottenPassword/ForgottenPassword.test.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { asyncScheduler, scheduled, throwError } from 'rxjs';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { ForgottenPassword, SendResetEmailDTO } from './ForgottenPassword';
+import { backendSrv } from '../../services/backend_srv';
+import { toAsyncOfResult } from '../../../features/query/state/DashboardQueryRunner/testHelpers';
+import { createFetchResponse } from '../../../../test/helpers/createFetchResponse';
+
+jest.mock('@grafana/runtime', () => ({
+  ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+  getBackendSrv: () => backendSrv,
+}));
+
+function getTestContext() {
+  jest.clearAllMocks();
+  const fetchMock = jest.spyOn(backendSrv, 'fetch');
+  const { rerender } = render(<ForgottenPassword />);
+
+  return { fetchMock, rerender };
+}
+
+describe('ForgottenPassword', () => {
+  describe('when mounted', () => {
+    it('then it should show input field', () => {
+      getTestContext();
+
+      expect(screen.getByRole('textbox')).toBeInTheDocument();
+    });
+
+    it('then it should show send button', () => {
+      getTestContext();
+
+      expect(screen.getByText(/send reset email/i)).toBeInTheDocument();
+    });
+
+    it('then it should show back to login link', () => {
+      getTestContext();
+
+      expect(screen.getByText(/back to login/i)).toBeInTheDocument();
+    });
+  });
+
+  describe('when user submits form', () => {
+    describe('and response is ok', () => {
+      it('then it should show success message', async () => {
+        const { fetchMock } = getTestContext();
+        fetchMock.mockImplementation(() =>
+          toAsyncOfResult(
+            createFetchResponse<SendResetEmailDTO>({ message: 'Success' })
+          )
+        );
+
+        await userEvent.type(screen.getByRole('textbox'), 'JaneDoe');
+        userEvent.click(screen.getByText(/send reset email/i));
```

Comment:
Same comments here about being more explicit with the matcher.

Reply:
Fixed!